src/client/Microsoft.Identity.Client/Internal/Requests/RequestBase.cs

```diff
@@ -98,20 +101,40 @@ public async Task<AuthenticationResult> RunAsync(CancellationToken cancellationT
                     LogMetricsFromAuthResult(authenticationResult, AuthenticationRequestParameters.RequestContext.Logger);
                     LogSuccessfulTelemetryToClient(authenticationResult, telemetryEventDetails, telemetryClients);
 
+#if NET6_0_OR_GREATER
```

Comment:
You can avoid having these compliation flags in the code by putting everything in a class, smth like `OpenTelemetryWrapper`